python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -315,5 +316,28 @@ def determinePsf(
 
         return psf, None
 
+    def _validatePsfCandidates(self, psfCandidateList, kernelSize):
```

Comment:
Do the `psfCandidates` in `psfCandidateList` have a well-defined (constant) size?  The python for loop in this function is adding an unnecessary performance hit if so.

Reply:
Good question: we don't enforce the psfCandidate size when constructing that list (and it's just a `list`, not a coherent data structure). But the primary usage gets them from `MakePsfCandidatesTask`, which has a single `config.kernelSize`. So, I think you're right that we can just assume they're all the same and only test the first one. I do hope that `piff` upstream implements an internal check as well, though.